### PR TITLE
Improve handling of http parameters and arguments for gmp

### PR DESCRIPTION
**Pull request by bjoernricks**
**Checklist**:

- [n/a] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: gsad/src/gsad_gmp.c [hunk 1]
```diff
@@ -1019,65 +815,51 @@ response_from_entity (gvm_connection_t *connection, credentials_t *credentials,
...
- * @param[in]  extra_attribs  Extra attributes for GMP GET command.
- * @param[out] response_data  Extra data return for the HTTP response.
- *
- * @return Enveloped XML object.
- */
 char *
-get_one (gvm_connection_t *connection, const char *type,
-         credentials_t *credentials, params_t *params, const char *extra_xml,
-         const char *extra_attribs, cmd_response_data_t *response_data)
+get_entity (gvm_connection_t *connection, const char *type,
```
Comment (timopollmeier) [inline gsad/src/gsad_gmp.c]: The get_entity function does not have any documentation comment.

File: gsad/src/gsad_gmp.c [hunk 2]
```diff
@@ -1124,209 +908,74 @@ get_one (gvm_connection_t *connection, const char *type,
...
  * @return Enveloped XML object.
  */
-static char *
-get_many (gvm_connection_t *connection, const char *type,
-          credentials_t *credentials, params_t *params, const char *extra_xml,
-          const char *extra_attribs, cmd_response_data_t *response_data)
+char *
+get_one (gvm_connection_t *connection, const char *type,
+         credentials_t *credentials, params_t *params, const char *extra_xml,
+         gmp_arguments_t *arguments, cmd_response_data_t *response_data)
```
Comment (timopollmeier) [inline gsad/src/gsad_gmp.c]: The documentation comment of get_one still refers to extra_attribs instead of arguments.

**Comment by timopollmeier** [comment]
Aside from the things I've noted above I've also noticed that the "Rows Per Page" setting is now ignored.
GSAd took care of this before but it may be a good idea to simplify it in that regard and move some of this to GVMd and the JS part of GSA.

File: gsad/src/gsad.c [hunk 3]
```diff
@@ -613,6 +611,8 @@ init_validator ()
   gvm_validator_add (validator, "name", "^[#-_[:alnum:], \\./]*$");
   gvm_validator_add (validator, "info_name", "(?s)^.*$");
   gvm_validator_add (validator, "info_type", "(?s)^.*$");
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply gsad/src/gsad.c]: I guess this was a mistake during rebasing from gsa-8 to master.

File: gsad/src/gsad_gmp.c (hunk 1 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: Fixed

File: gsad/src/gsad.c [hunk 4]
```diff
@@ -2104,7 +2104,6 @@ exec_gmp_get (http_connection_t *con, gsad_connection_info_t *con_info,
   ELSE (get_config_family)
   ELSE (get_config_nvt)
   ELSE (ping)
-  ELSE (sync_config)
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply gsad/src/gsad.c]: It isn't used in GSA thus I removed it. If we need it in the feature we can add it again.

File: gsad/src/gsad_gmp.c [hunk 5]
```diff
@@ -1095,10 +1095,8 @@ get_many (gvm_connection_t *connection, const char *type,
           credentials_t *credentials, params_t *params, const char *extra_xml,
           gmp_arguments_t *arguments, cmd_response_data_t *response_data)
 {
-  GString *type_many; /* The plural form of type */
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: Fixed

File: gsad/src/gsad_gmp.c (hunk 2 above)
Comment (bjoernricks) [reply gsad/src/gsad_gmp.c]: Fixed

**Comment by bjoernricks** [comment]
> Aside from the things I've noted above I've also noticed that the "Rows Per Page" setting is now ignored.
GSAd took care of this before but it may be a good idea to simplify it in that regard and move some of this to GVMd and the JS part of GSA.

Yes the web server should be only a proxy between GSA and gvmd and therefore all request preparation and params should be done in the js code now.

File: gsad/src/gsad_gmp_arguments.c [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+void
+gmp_arguments_ref (gmp_arguments_t *arguments)
+{
+  g_hash_table_ref (arguments);
+}
+
+void
+gmp_arguments_free (gmp_arguments_t *arguments)
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply gsad/src/gsad_gmp_arguments.c]: Oh I thinks this is a leftover. I couldn't agree on using referencing or freeing with myself ;-) I'll change it in a future PR. Thanks a lot!

**Pull request merged**